packages/data/src/api/dynamicNftMetadata/dynamicNftMetadata.test.ts

```diff
@@ -0,0 +1,62 @@
+import { dynamicNftMetadata } from './dynamicNftMetadata';
+import { DynamicNftMetadataResult } from './dynamicNftMetadata.types';
+
+import { GraphQLClient } from 'graphql-request';
+import { dynamicNftMetadataMockInactive, dynamicNftMetadataMockActive } from './dynamicNftMetadata.mock';
+import { Network } from '@mintbase-js/sdk';
+
+jest.mock('graphql-request');
+
+const props = {
+  contractAddress: 'dynamic.mintbase2.near',
+  internalMetadataId: 0,
+  network: 'mainnet' as Network,
+};
+
+describe('dynamicNftMetadata', () => {
+  jest.spyOn(console, 'error').mockImplementation(() => null);
+  afterAll(() => {
+    jest.resetAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return active metadata', async () => {
+    (GraphQLClient as jest.Mock).mockImplementationOnce(() => ({
+      request: (): Promise<DynamicNftMetadataResult> => Promise.resolve(dynamicNftMetadataMockActive),
+    }));
+
+    const result = await dynamicNftMetadata({ ...props, isActive: true });
+
+    expect(result?.data).toBe(
+      dynamicNftMetadataMockActive,
+    );
+  });
+
+  it('should return inactive metadata', async () => {
+    (GraphQLClient as jest.Mock).mockImplementationOnce(() => ({
+      request: (): Promise<DynamicNftMetadataResult> => Promise.resolve(dynamicNftMetadataMockInactive),
+    }));
+
+    const result = await dynamicNftMetadata({ ...props, isActive: false });
+
+    expect(result?.data).toBe(
+      dynamicNftMetadataMockInactive,
+    );
+  });
+
+  it('should handle errors', async () => {
+    const errMessage = 'exploded';
+    (GraphQLClient as jest.Mock).mockImplementationOnce(() => ({
+      request: (): Promise<DynamicNftMetadataResult> => Promise.reject(new Error(errMessage)),
+    }));
+
+    const call = await dynamicNftMetadata({...props, isActive: true});
+
+    expect(call).toStrictEqual({ error: errMessage });
```

Comment:
Here, I think we should be mocking the internal `fetchGraphQl` to actually test that `dynamicNftMetadata` is handling errors the way we expect and the assertion should look like:

```suggestion
    expect(call).toStrictEqual({ error: `Error fetching store nfts, ${errMessage}` });
```

Because that's what the function does with errors. 

Otherwise, I don't believe this test serves any non-trivial purpose.

Reply:
Hm, the method `GraphQlClient.request` is being used inside of `fetchGraphQl`, so this is indeed a bit spurious. Just dug into `parseData`, the `Error fetching store NFTs` is simply logged, but not returned. it returns the unchanged error string or object. To change that would require rewriting the tests for the whole module.